extensions/VisualEditor/lib/ve/lib/jquery.client/jquery.client.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * User-agent detection
+ */
+( function ( $ ) {
+
+	/* Private Members */
+
+	/**
+	 * @var {Object} profileCache Keyed by userAgent string,
+	 * value is the parsed $.client.profile object for that user agent.
+	 */
+	var profileCache = {};
+
+	/* Public Methods */
+
+	$.client = {
+
+		/**
+		 * Get an object containing information about the client.
+		 *
+		 * @param {Object} nav An object with atleast a 'userAgent' and 'platform' key.
+		 * Defaults to the global Navigator object.
+		 * @returns {Object} The resulting client object will be in the following format:
+		 *  {
+		 *   'name': 'firefox',
+		 *   'layout': 'gecko',
+		 *   'layoutVersion': 20101026,
+		 *   'platform': 'linux'
+		 *   'version': '3.5.1',
+		 *   'versionBase': '3',
+		 *   'versionNumber': 3.5,
+		 *  }
+		 */
+		profile: function ( nav ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has a complexity of 27 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)